lobster/report.py

```diff
@@ -193,7 +200,39 @@ def load_report(self, filename):
                                               3)
 
                 self.items[item.tag.key()] = item
-                self.coverage[item.level]["items"] += 1
+                self.coverage[item.level].items += 1
                 if item.tracing_status in (Tracing_Status.OK,
                                            Tracing_Status.JUSTIFIED):
-                    self.coverage[item.level]["ok"] += 1
+                    self.coverage[item.level].ok += 1
+
+    def validate_indicated_schema(self, data, loc):
+        supported_schema = {
+            "lobster-report": set([2]),
+        }
+        if data["schema"] not in supported_schema:
+            self.mh.error(loc, "unknown schema kind %s" % data["schema"])
+        if data["version"] not in supported_schema[data["schema"]]:
+            self.mh.error(loc,
+                          "version %u for schema %s is not supported" %
+                          (data["version"], data["schema"]))
+
+    def validate_basic_structure_of_lobster_file(self, data, loc):
+        if not isinstance(data, dict):
+            self.mh.error(loc, "parsed json is not an object")
+        for rkey in ("schema", "version", "generator",
+                     "levels", "policy", "matrix"):
+            if rkey not in data:
+                self.mh.error(loc,
+                              "required top-levelkey %s not present" % rkey)
+            if rkey in ("levels", "matrix"):
+                if not isinstance(data[rkey], list):
+                    self.mh.error(loc, "%s is not an array" % rkey)
+            elif rkey == "policy":
+                if not isinstance(data[rkey], dict):
+                    self.mh.error(loc, "policy is not an object")
+            elif rkey == "version":
+                if not isinstance(data[rkey], int):
+                    self.mh.error(loc, "version is not an integer")
+            else:
+                if not isinstance(data[rkey], str):
+                    self.mh.error(loc, "%s is not a string" % rkey)
```

Comment:
I would suggest instead of checking in if elif elase block we can pass dict which contain key and data type and in single if block we can check data type is correct or not if not then we will though an error like below 
{"schema : str,
"version" : int,}
if not isntance(data[key], type)
    self.mh.error((loc, "%s is not a type" % type)

Reply:
Made the change